File: SwiftHelperBackend/SwiftHelperBackend.DAL/Models/PlanStageEntry.cs
using SwiftHelperBackend.DAL.Models.Enums;

namespace SwiftHelperBackend.DAL.Models;

public class PlanStageEntry: IdBase
{
    public string Title { get; set; } = null!;
    public StagePriority Priority { get; set; } = StagePriority.Empty;
    public StageStatus Status { get; set; } = StageStatus.New;
    public string Data { get; set; } = null!;
    public DateTime CreatedDate { get; set; }
    
    public string UserId { get; set; } = null!;
    public ApplicationUser User { get; set; } = null!;
    
    public long PlanEntryId { get; set; }
    public PlanEntry PlanEntry { get; set; } = null!;
}

File: flutter_new_app/lib/models/api/plan_stage_entry.dart
import 'package:flutter_new_app/models/enums/stage_priority.dart';
import 'package:flutter_new_app/models/enums/stage_status.dart';

class PlanStageEntry {
  final num id;
  final String title;
  final String data;
  final StagePriority priority;
  final StageStatus status;
  final DateTime createdDate;
  final num planEntryId;

  PlanStageEntry(
      {required this.id,
      required this.title,
      required this.data,
      required this.priority,
      required this.status,
      required this.createdDate,
      required this.planEntryId});

  factory PlanStageEntry.fromJson(Map<String, dynamic> json) {
    return PlanStageEntry(
        id: json['id'] as num,
        title: json['title'] as String,
        data: json['data'] as String,
        priority: StagePriority.values[json['priority'] as int],
        status: StageStatus.values[json['status'] as int],
        createdDate: DateTime.parse(json['createdDate'] as String),
        planEntryId: json['planEntryId'] as num);
  }

  Map<String, dynamic> toJson() => {
        'id': id as int,
        'title': title,
        'data': data,
        'priority': priority.index,
        'status': status.index,
        'createdDate': formattingDate(createdDate ?? DateTime.now()),
        'planEntryId': planEntryId as int
      };

  String formattingDate(DateTime date) =>
      '${date.year.toString()}-${date.month.toString().length == 1 ? '0${date.month}' : date.month.toString()}-${date.day.toString()}';
}
